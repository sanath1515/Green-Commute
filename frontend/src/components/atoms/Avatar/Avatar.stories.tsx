import React from "react";
import { Avatars } from "./Avatar";
import img from "./women.jpeg";

export default {
  title: "Atoms/Avatars",
  component: Avatars,
};

const Template = (args) => <Avatars {...args} />;

export const Letter = Template.bind({});

export const Image = Template.bind({});

export const Number = Template.bind({});

Image.args = {
  alt: "A",
  src: img,
};

Letter.args = {
  alt: "A",
  src: "img",
  valid: true,
};

Number.args = {
  alt: "1",
  src: "A",
  valid: true,
};
