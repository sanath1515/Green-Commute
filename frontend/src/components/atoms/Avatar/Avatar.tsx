import { Avatar } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

type AvatarProps = {
  alt?: string;
  src?: string;
  variant?: "circular" | "rounded" | "square";
  className?: string;
  valid?: boolean;
};
const useStyles = makeStyles({
  root: {
    backgroundColor: "grey",
  },
});

export const Avatars = (props: AvatarProps) => {
  const classes = useStyles();
  return (
    <Avatar
      alt={props.alt}
      src={props.src}
      variant={props.variant}
      className={props.className}
      classes={{ root: props.valid ? classes.root : "" }}
    />
  );
};
