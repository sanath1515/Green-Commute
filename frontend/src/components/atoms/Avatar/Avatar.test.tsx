/** * @jest-environment jsdom */
import { render } from "@testing-library/react";
import { Avatars } from "./Avatar";

describe("Avatar test", () => {
  const src =
    "https://cdn.pixabay.com/photo/2015/12/01/20/28/road-1072823__340.jpg";
  it("should render letter", () => {
    const { container } = render(<Avatars alt="a" />);
    expect(container.firstChild).toBeTruthy();
  });

  it("should render the image", () => {
    const { container } = render(<Avatars src={src} />);
    expect(container.firstChild).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { container } = render(<Avatars src={src} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
