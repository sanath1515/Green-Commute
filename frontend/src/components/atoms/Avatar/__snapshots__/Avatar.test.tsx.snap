// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar test should match the snapshot 1`] = `
<div
  class="MuiAvatar-root MuiAvatar-circular"
>
  <img
    class="MuiAvatar-img"
    src="https://cdn.pixabay.com/photo/2015/12/01/20/28/road-1072823__340.jpg"
  />
</div>
`;
